static/app/views/issueList/container.tsx

```diff
@@ -1,33 +1,56 @@
-import {Component} from 'react';
+import React, {Component} from 'react';
 import DocumentTitle from 'react-document-title';
 
 import LightWeightNoProjectMessage from 'app/components/lightWeightNoProjectMessage';
 import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
-import {Organization} from 'app/types';
+import GroupStore from 'app/stores/groupStore';
+import {Organization, Project} from 'app/types';
+import {callIfFunction} from 'app/utils/callIfFunction';
 import withOrganization from 'app/utils/withOrganization';
+import SampleEventAlert from 'app/views/organizationGroupDetails/sampleEventAlert';
 
 type Props = {
   organization: Organization;
+  projects: Project[];
 };
 
 class IssueListContainer extends Component<Props> {
+  state: {
+    showSampleEventBanner: boolean;
+  } = {
+    showSampleEventBanner: false,
+  };
+
   getTitle() {
     return `Issues - ${this.props.organization.slug} - Sentry`;
   }
 
+  listener = GroupStore.listen(() => this.onGroupChange(), undefined);
   render() {
     const {organization, children} = this.props;
-
     return (
       <DocumentTitle title={this.getTitle()}>
-        <GlobalSelectionHeader>
-          <LightWeightNoProjectMessage organization={organization}>
-            {children}
-          </LightWeightNoProjectMessage>
-        </GlobalSelectionHeader>
+        <React.Fragment>
+          {this.state.showSampleEventBanner && <SampleEventAlert />}
+          <GlobalSelectionHeader>
+            <LightWeightNoProjectMessage organization={organization}>
+              {children}
+            </LightWeightNoProjectMessage>
+          </GlobalSelectionHeader>
+        </React.Fragment>
       </DocumentTitle>
     );
   }
+
+  onGroupChange() {
+    this.setState({
+      showSampleEventBanner: GroupStore.getAllItemIds().length === 1,
```

Comment:
@Neo-Zhixing Does this mean we show the sample event banner if there is exactly 1 issue? What if the only issue was not a sample error? I thought we could just use the `first_event` field on the project to determine this.

Reply:
Nope.
```
{this.state.showSampleEventBanner && <SampleEventAlert />}
```
This means that we will only add `<SampleEventAlert />` when there is exactly 1 issue. And `<SampleEventAlert />` will only render itself when `first_event` is false.

This is because `first_event` is also false when there are no events at all. If we only look at `first_event`, the end result is that we will display the banner on the listing page even if we're showing the empty state.

Reply:
Makes sense!